zerver/tests/test_subs.py

```diff
@@ -498,6 +499,20 @@ def get_streams(group: DefaultStreamGroup) -> List[Stream]:
         do_deactivate_stream(streams_to_remove[0])
         self.assertEqual(get_streams(default_stream_groups[0]), streams_to_keep)
 
+    def test_deactivate_stream_marks_messages_as_read(self) -> None:
+        hamlet = self.example_user("hamlet")
+        cordelia = self.example_user("cordelia")
+        stream = self.make_stream('new_stream')
+        self.subscribe(hamlet, stream.name)
+        self.subscribe(cordelia, stream.name)
+        self.send_stream_message(cordelia, stream.name)
+        usermessage = most_recent_usermessage(cordelia)
```

Comment:
I think we want a slightly stronger test than this.  It's not completely clear what we're checking with Hamlet here.

Also, since we're using the big hammer here of obliterating all the UserMessages in the stream, I think the biggest bug to protect against is accidentally deleting messages outside of the stream, so we want a test that, say, Cordelia have a couple unread messages on Denmark, and then deactivating "new_stream" doesn't make those messages read.

Reply:
Ah, this was actually supposed to be hamlet sending the message, and cordelia being the "receiver". So I fixed that, and added the additional verification you mentioned, good idea!